clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderProducerCallback.java

```diff
@@ -144,30 +144,13 @@ public void onCompletion(RecordMetadata recordMetadata, Exception e) {
           producedRecordSize =
               Math.max(0, recordMetadata.serializedKeySize()) + Math.max(0, recordMetadata.serializedValueSize());
         } else {
-          int schemaId = AvroProtocolDefinition.CHUNK.getCurrentProtocolVersion();
-          for (int i = 0; i < chunkedValueManifest.keysWithChunkIdSuffix.size(); i++) {
-            ByteBuffer chunkKey = chunkedValueManifest.keysWithChunkIdSuffix.get(i);
-            ByteBuffer chunkValue = valueChunks[i];
-
-            Put chunkPut = instantiateChunkPut();
-            chunkPut.putValue = chunkValue;
-            chunkPut.schemaId = schemaId;
-
-            LeaderProducedRecordContext producedRecordForChunk =
-                LeaderProducedRecordContext.newChunkPutRecord(ByteUtils.extractByteArray(chunkKey), chunkPut);
-            producedRecordForChunk.setProducedOffset(-1);
-            ingestionTask.produceToStoreBufferService(
-                sourceConsumerRecord,
-                producedRecordForChunk,
-                subPartition,
-                kafkaUrl,
-                beforeProcessingRecordTimestamp);
-            producedRecordNum++;
-            producedRecordSize += chunkKey.remaining() + chunkValue.remaining();
+          producedRecordSize += produceChunksToStoreBufferService(chunkedValueManifest, valueChunks, false);
+          producedRecordNum += chunkedValueManifest.keysWithChunkIdSuffix.size();
+          if (chunkedRmdManifest != null) {
```

Comment:
Based on the new logic in VeniceWriter, `chunkedRmdManifest` can't be `null` when `chunkedValueManifest` is not `null`, right?

Reply:
chunkedRmdManifest and rmdChunks can be null. When there is no RMD chunk being produced, these two fields will be null.

Reply:
You are right.